build-tools/packages/build-cli/src/commands/updates/packageTypes.ts

```diff
@@ -0,0 +1,157 @@
+			description:
+				"The types flag is used to specify the type of release when updating a package.json file. It accepts a custom enumeration called UpdatePackageJsonEnum that defines different release types.",
+			parse: async (input) => {
+				if (isUpdatePackageJsonTypes(input)) {
+					return input;
+				}
+				throw new Error(`Invalid release type: ${input}`);
+			},
+		})(),
+		...PackageCommand.flags,
+	};
+
+	private readonly packageTypesUpdated: string[] = [];
+
+	protected async processPackage(pkg: Package): Promise<void> {
+		// Define paths to package.json and api-extractor.json
+		const packageJsonFilePath = path.join(pkg.directory, "package.json");
+		const apiExtractorConfigFilePath = path.join(pkg.directory, "api-extractor.json");
+
+		// Check if api-extractor.json exists in the package directory
+		const apiExtractorConfigExists = fs.existsSync(apiExtractorConfigFilePath);
+
+		try {
+			// Read the content of the package.json file
+			const packageJsonContent = fs.readFileSync(packageJsonFilePath, "utf-8");
+
+			// Parse the JSON content to an object
+			const packageJson = JSON.parse(packageJsonContent);
+
+			if (apiExtractorConfigExists) {
+				// Read the content of api-extractor.json
+				const apiExtractorConfigContent = fs.readFileSync(
+					apiExtractorConfigFilePath,
+					"utf-8",
+				);
+
+				// Check if DTSROLLUP config exist in api-extractor.json
+				const isDtsRollupEnabled = apiExtractorConfigContent.includes(
+					ApiExtractorConfig.DTSROLLUP,
+				);
+				// Check if ALPHATRIMMED config exist in api-extractor.json
+				const isAlphaTrimmed = apiExtractorConfigContent.includes(
+					ApiExtractorConfig.ALPHATRIMMED,
+				);
+				// Check if BETATRIMMED config exist in api-extractor.json
+				const isBetaTrimmed = apiExtractorConfigContent.includes(
+					ApiExtractorConfig.BETATRIMMED,
+				);
+
+				if (isDtsRollupEnabled) {
+					const packageJsonUpdatePath = path.join(pkg.directory, "package.json");
+
+					// Extract the package name from the package directory
+					const directorySegments: string[] = pkg.directory.split("/");
+					const unscopedPackageName = directorySegments[directorySegments.length - 1];
+
+					// Determine the value for the packageJson.types field based on flags
+					if (this.flags.types === UpdatePackageJsonEnum.BETA && isBetaTrimmed) {
+						packageJson.types = `dist/${unscopedPackageName}-beta.d.ts`;
+						this.packageTypesUpdated.push(pkg.name);
+					} else if (this.flags.types === UpdatePackageJsonEnum.ALPHA && isAlphaTrimmed) {
+						packageJson.types = `dist/${unscopedPackageName}-alpha.d.ts`;
+						this.packageTypesUpdated.push(pkg.name);
+					} else if (this.flags.types === UpdatePackageJsonEnum.DEFAULT) {
+						packageJson.types = "dist/index.d.ts";
+						this.packageTypesUpdated.push(pkg.name);
+					}
+
+					// Serialize the updated package.json without modifying the formatting
+					const updatedPackageJsonContent = JSON.stringify(
```

Comment:
If you have to write the file, definitely use the function Craig mentioned. I'm pretty sure it writes the file using the prettier config so you get automatically correct formatting without doing anything.